daemon/images/image_list.go

```diff
@@ -75,6 +77,22 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 		return nil, err
 	}
 
+	err = opts.Filters.WalkValues("until", func(value string) error {
+		ts, err := timetypes.GetTimestamp(value, time.Now())
+		if err != nil {
+			return err
+		}
+		seconds, nanoseconds, err := timetypes.ParseTimestamps(ts, 0)
+		if err != nil {
+			return err
+		}
+		untilFilter = time.Unix(seconds, nanoseconds)
+		return nil
+	})
```

Comment:
The callback function will be called for each `until` condition in the filter. There may be more than one `until` condition in the request. When there are multiple `until` conditions, which timestamp is used when filtering images? [It can't be last-one-wins.](https://github.com/moby/moby/pull/44503)

Reply:
That is a good point. I will make a modification that will, out of all the timestamps given by before and until filters, choose the oldest timestamp. This way we can get rid of the `untilFilter` variable as well. I should be able to implement and push this modification in next 3 - 4 days.